src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,146 @@
+
+        return genericMessage("RecommendationRequest with id %s deleted".formatted(id));
+    }
+
+    /**
+     * Any admin can update a single Recommendation Request
+     * 
+     * @param id       the id of the Recommendation Request to update
+     * @param incoming the updated Recommendation Request
+     * @return the updated Recommendation Request object
+     */
+    @Operation(summary = "An admin can update a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/admin")
+    public RecommendationRequest updateRecommendationRequestAsAdmin(
+        @Parameter(name = "id") @RequestParam Long id,
+        @RequestBody @Valid RecommendationRequest incoming) {
+
+        RecommendationRequest recommendationRequest = 
+            recommendationRequestRepository
+                .findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequest.setProfessorName(incoming.getProfessorName());
+        recommendationRequest.setProfessorEmail(incoming.getProfessorEmail());
+        recommendationRequest.setRequesterName(incoming.getRequesterName());
+        recommendationRequest.setRecommendationTypes(incoming.getRecommendationTypes());
+        recommendationRequest.setDetails(incoming.getDetails());
+        recommendationRequest.setStatus(incoming.getStatus());
+        recommendationRequest.setSubmissionDate(incoming.getSubmissionDate());
+        recommendationRequest.setCompletionDate(incoming.getCompletionDate());
+
+        recommendationRequestRepository.save(recommendationRequest);
+
+        return recommendationRequest;
+    }
+
+    /**
+     * The user who posted a RecommendationRequest can update their RecommendationRequest
+     * 
+     * @param id       the id of the Recommendation Request to update
+     * @param incoming the updated Recommendation Request
+     * @return the updated Recommendation Request object
+     */
+    @Operation(summary = "User can update their RecommendationRequest")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PutMapping("")
+    public RecommendationRequest updateRecommendationRequestAsUser( 
+        @Parameter(name = "id") @RequestParam Long id,
+        @RequestBody @Valid RecommendationRequest incoming) {
+
+        User currentUser = getCurrentUser().getUser(); 
+        RecommendationRequest recommendationRequest = 
+            recommendationRequestRepository
+                .findByIdAndUser(id, currentUser)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequest.setProfessorName(incoming.getProfessorName());
+        recommendationRequest.setProfessorEmail(incoming.getProfessorEmail());
+        recommendationRequest.setRecommendationTypes(incoming.getRecommendationTypes());
+        recommendationRequest.setDetails(incoming.getDetails());
```

Comment:
Honestly this is the ONLY field a user should be able to update.

And the only thing a professor should be able to update is the status, and possibly be able to enter some professorComments.   (Which is something we don't have in the database right now but probably should.. though that can be a future PR).